File: SharedPreferences/app/src/main/java/com/example/sharedpreferences/MainActivity.java
package com.example.sharedpreferences;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import com.example.sharedpreferences.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private String email,pass;
    public SharedPreferences sharedPreferences;
    private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        sharedPreferences = getSharedPreferences(Constant.PREFERENCE_NAME, Context.MODE_PRIVATE);
        binding.saveButton.setOnClickListener(this);
        binding.go.setOnClickListener(this);
    }
    public void onClick(View view){
        switch (view.getId()){
            case R.id.saveButton:
                email = binding.email.getText().toString();
                pass = binding.pass.getText().toString();
                if (email.isEmpty()){
                    binding.email.setError(getString(R.string.enteremail));
                }
                else if (pass.isEmpty()){
                    binding.pass.setError(getString(R.string.enterpass));
                }
                else{
                    saveData();
                }
                break;
            case R.id.go:
                startActivity(new Intent(MainActivity.this,SideActivity.class));
        }
    }
    public void saveData(){
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(Constant.EMAIL,email);
        editor.putString(Constant.PASS,pass);
        editor.apply();
        binding.email.setText("");
        binding.pass.setText("");
    }
}

File: ToDoList/app/src/main/java/com/example/to_dolist/ui/MainActivity.kt
package com.example.to_dolist.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import com.example.to_dolist.R
import com.example.to_dolist.data.localDatabase.TodoListDatabase
import com.example.to_dolist.databinding.ActivityMainBinding
import com.example.to_dolist.utils.DataState
import com.example.to_dolist.viewModel.TodoListViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import timber.log.Timber

@AndroidEntryPoint
class MainActivity :AppCompatActivity(){
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}